journal/journal.go

```diff
@@ -50,19 +50,35 @@ const (
 	PriDebug
 )
 
-var conn net.Conn
+var (
+	conn *net.UnixConn
+	// This can be overridden at build-time:
+	// https://github.com/golang/go/wiki/GcToolchainTricks#including-build-information-in-the-executable
+	journalSocket = "/run/systemd/journal/socket"
+)
 
 func init() {
-	var err error
-	conn, err = net.Dial("unixgram", "/run/systemd/journal/socket")
+	autobind, err := net.ResolveUnixAddr("unixgram", "")
+	if err != nil {
+		return
+	}
+	conn, err = net.ListenUnixgram("unixgram", autobind)
 	if err != nil {
 		conn = nil
+		return
 	}
 }
 
 // Enabled returns true if the local systemd journal is available for logging
 func Enabled() bool {
-	return conn != nil
+	if conn == nil {
+		return false
+	}
+	_, err := net.ResolveUnixAddr("unixgram", journalSocket)
+	if err == nil {
```

Comment:
Note to self: this is a tautological condition due to `ResolveUnixAddr` implementation, and can be removed.